src/cljs/rems/administration/create_catalogue_item.cljs

```diff
@@ -228,31 +231,40 @@
    {:on-click #(dispatch! "/#/administration/catalogue-items")}
    (text :t.administration/cancel)])
 
-(defn- save-catalogue-item-button []
+(defn- save-catalogue-item-button [on-click]
   (let [form @(rf/subscribe [::form])
         request (build-request form)]
     [:button.btn.btn-primary
-     {:on-click #(rf/dispatch [::create-catalogue-item request])
+     {:on-click #(on-click request)
       :disabled (nil? request)}
      (text :t.administration/save)]))
 
 (defn create-catalogue-item-page []
-  (let [loading? (rf/subscribe [::loading?])]
-    [:div
-     [administration-navigator-container]
-     [:h2 (text :t.administration/create-catalogue-item)]
-     [collapsible/component
-      {:id "create-catalogue-item"
-       :title (text :t.administration/create-catalogue-item)
-       :always [:div
-                (if @loading?
-                  [:div#catalogue-item-loader [spinner/big]]
-                  [:div#catalogue-item-editor
-                   [catalogue-item-title-field]
-                   [catalogue-item-workflow-field]
-                   [catalogue-item-resource-field]
-                   [catalogue-item-form-field]
-
-                   [:div.col.commands
-                    [cancel-button]
-                    [save-catalogue-item-button]]])]}]]))
+  (let [loading? (rf/subscribe [::loading?])
+        {:keys [on-pending on-success on-error state-atom] :as modal-opts} (status-modal/status-modal-state-handling
```

Comment:
The name could be more descriptive. How about `status-modal/create`, `status-modal/create-state`, `status-modal/init` or something?